examples/nodejs/get-set-batch-perf-test/perf-test.ts

```diff
@@ -0,0 +1,239 @@
+    const setBatchPromise = momento.setBatch(this.cacheName, items);
+
+    void setBatchPromise.then(() => {
+      const setBatchDuration = getElapsedMillis(setBatchStartTime);
+      context.setBatchLatencies.recordValue(setBatchDuration);
+    });
+    context.totalItemSizeBytes += setConfig.batchSize * setConfig.itemSizeBytes;
+    await setBatchPromise;
+  }
+
+  private async sendGetBatchRequests(
+    momento: CacheClient,
+    context: PerfTestContext,
+    getConfig: GetSetConfig
+  ): Promise<void> {
+    const keys = Array.from({length: getConfig.batchSize}, (_, i) => `key-${i}`);
+    const getBatchStartTime = process.hrtime();
+    const getBatchPromise = momento.getBatch(this.cacheName, keys);
+    void getBatchPromise.then(() => {
+      const getBatchDuration = getElapsedMillis(getBatchStartTime);
+      context.getBatchLatencies.recordValue(getBatchDuration);
+    });
+    context.totalItemSizeBytes += getConfig.batchSize * getConfig.itemSizeBytes;
+    await getBatchPromise;
+  }
+
+  private asyncSet(
+    momento: CacheClient,
+    context: PerfTestContext,
+    key: string,
+    value: string
+  ): Promise<CacheSet.Response> {
+    const startTime = process.hrtime();
+    const setPromise = momento.set(this.cacheName, key, value);
+    void setPromise.then(() => {
+      const setDuration = getElapsedMillis(startTime);
+      context.asyncSetLatencies.recordValue(setDuration);
+    });
+    return setPromise;
+  }
+
+  private asyncGet(momento: CacheClient, context: PerfTestContext, key: string): Promise<CacheSet.Response> {
+    const startTime = process.hrtime();
+    const getPromise = momento.get(this.cacheName, key);
+    void getPromise.then(() => {
+      const getDuration = getElapsedMillis(startTime);
+      context.asyncGetLatencies.recordValue(getDuration);
+    });
+    return getPromise;
+  }
+
+  private async ensureCacheIsPopulated(momento: CacheClient, getConfig: GetSetConfig) {
+    const keys = Array.from({length: getConfig.batchSize}, (_, i) => `key-${i}`);
+    const values = Array.from({length: getConfig.batchSize}, () => 'x'.repeat(getConfig.itemSizeBytes));
+
+    for (let i = 0; i < keys.length; i++) {
+      await momento.setIfAbsent(this.cacheName, keys[i], values[i]);
+    }
+  }
+}
+
+async function main(perfTestOptions: PerfTestOptions, testConfiguration: PerfTestConfiguration): Promise<void> {
+  const perfTest = new PerfTest(perfTestOptions, testConfiguration);
+  await perfTest.run();
+}
+
+const batchSizeOptions = [5, 10, 100, 500, 1_000, 5_000, 10_000];
+const itemSizeOptions = [10, 100, 1024, 1024 * 10, 1024 * 100, 1024 * 1024];
+
+const testConfiguration: PerfTestConfiguration = {
+  minimumRunDurationSecondsForTests: 5,
+  sets: generateConfigurations(batchSizeOptions, itemSizeOptions),
+  gets: generateConfigurations(batchSizeOptions, itemSizeOptions),
+};
+
+function generateConfigurations(batchSizes: number[], itemSizes: number[]): GetSetConfig[] {
```

Comment:
i like this function! we might end up having to filter some permutations out for `set` though if we know that they will cross the item size threshold. but this is a great starting point.